compiler/src/iree/compiler/Codegen/Common/GenericVectorization.cpp

```diff
@@ -325,6 +325,14 @@ class GenericVectorizationPass final
   void runOnOperation() override;
 };
 
+/// Converts from iree_compiler::VscaleRange to vector::VscaleRange.
+static std::optional<vector::VscaleRange>
+toVectorVscaleRange(std::optional<iree_compiler::VscaleRange> vscaleRange) {
+  if (!vscaleRange.has_value())
+    return std::nullopt;
+  return vector::VscaleRange{vscaleRange->min, vscaleRange->max};
+}
```

Comment:
I was confused that why can't we just use `vector::VscaleRagnge`, then I found that it's added to upstream recently (https://github.com/llvm/llvm-project/commit/9b06e25e73470612d14f0e1e18fde82f62266216). Can you help remove the IREE version, and switch to use upstream version? It can be done in a follow-up.

Reply:
I'll do this in a follow up PR :+1: